### PR TITLE
TRINITY-47 Normalised the favourites url for current user

### PR DESCRIPTION
**Pull request by adrianfish**
https://sakaiproject.atlassian.net/browse/TRINITY-47

Also, used the spelling of favourites used by the vast majority of
english speakers in the world. We may have lost our empire, but we still
have our favourites, ha ha ha!

### CODE REVIEW

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/SitesController.java [hunk 1]
```diff
@@ -175,9 +173,7 @@
         }).collect(Collectors.toList());
 
         Map<String, List<Map<String, Object>>> data = new HashMap<>();
-        data.put("favorites", sitesList);
+        data.put("favourites", sitesList);
```
Comment (ottenhoff) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/SitesController.java:176]: 🤮

Comment (adrianfish) [reply webapi/src/main/java/org/sakaiproject/webapi/controllers/SitesController.java:160]: I also favour the phonetics of favour. Favour sounds different to favor. To me favor sounds like it should be pronounced in an italian or spanish way, like sabor. Which is funny, seeing as sabor means flavour.

Comment (adrianfish) [reply webapi/src/main/java/org/sakaiproject/webapi/controllers/SitesController.java:160]: So, you be sick all you want. Enjoy the flavour of that favour.

Comment (adrianfish) [reply webapi/src/main/java/org/sakaiproject/webapi/controllers/SitesController.java:160]: :)

Comment (adrianfish) [reply webapi/src/main/java/org/sakaiproject/webapi/controllers/SitesController.java:160]: However, if you insist, I will revert it to favorites.

**Review by stetsche — APPROVED** [review]
Makes sense to me if we really just need the current users favorites.

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/SitesController.java [hunk 2]
```diff
@@ -103,48 +104,29 @@
...
+        String userId = checkSakaiSession().getUserId();
 
-    private Map<String, List<Map<String, Object>>> getUserFavoriteSitesWithPagesMap(String userId) throws UserNotDefinedException  {
+        ResourceProperties resourceProperties = preferencesService.getPreferences(userId)
+            .getProperties(PreferencesService.SITENAV_PREFS_KEY);
+        List<String> siteIds = resourceProperties.getPropertyList("order");
 
-        ResourceProperties resourceProperties = preferencesService.getPreferences(userId).getProperties(org.sakaiproject.user.api.PreferencesService.SITENAV_PREFS_KEY);
-        List<String> propertyList = resourceProperties.getPropertyList("order");
+        if (siteIds == null) return Collections.emptyMap();
```
Comment (bgarciaentornos) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/SitesController.java:116]: Please, add brackets here

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/SitesController.java [hunk 3]
```diff
@@ -103,48 +104,29 @@
...
+                Site site = siteService.getSite(siteId);
                 siteMap.put("id", site.getId());
                 siteMap.put("title", site.getTitle());
                 siteMap.put("url", site.getUrl());
                 siteMap.put("type", site.getType());
                 List<Map<String, Object>> pageList = site.getOrderedPages().stream().map(page -> {
                     Map<String, Object> pageMap = new HashMap<>();
                     List<ToolConfiguration> toolList = page.getTools();
-                    if(toolList != null && toolList.size() == 1 ) {
+                    if (toolList.size() == 1 ) {
```
Comment (bgarciaentornos) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/SitesController.java:129]: Is it safe to remove the != null condition?

**Review by bgarciaentornos — CHANGES_REQUESTED** [review]

**Comment by mpellicer** [comment]
Incorporated in https://github.com/sakaiproject/sakai/pull/10290

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/SitesController.java (hunk 3 above)
Comment (adrianfish) [reply webapi/src/main/java/org/sakaiproject/webapi/controllers/SitesController.java:129]: Yes.

**Comment by adrianfish** [comment]
@stetsche Sorry I didn't get back to you in time. Thanks for folding those changes into your pr.

**Comment by stetsche** [comment]
@adrianfish no worries ;)

**Pull request closed without merging**